store-implementation/accumulo-store/src/main/java/uk/gov/gchq/gaffer/accumulostore/MiniAccumuloStore.java

```diff
@@ -0,0 +1,274 @@
+import uk.gov.gchq.gaffer.operation.impl.add.AddElements;
+import uk.gov.gchq.gaffer.operation.impl.compare.Max;
+import uk.gov.gchq.gaffer.operation.impl.compare.Min;
+import uk.gov.gchq.gaffer.operation.impl.compare.Sort;
+import uk.gov.gchq.gaffer.operation.impl.export.GetExports;
+import uk.gov.gchq.gaffer.operation.impl.export.set.ExportToSet;
+import uk.gov.gchq.gaffer.operation.impl.export.set.GetSetExport;
+import uk.gov.gchq.gaffer.operation.impl.function.Aggregate;
+import uk.gov.gchq.gaffer.operation.impl.function.Filter;
+import uk.gov.gchq.gaffer.operation.impl.function.Transform;
+import uk.gov.gchq.gaffer.operation.impl.generate.GenerateElements;
+import uk.gov.gchq.gaffer.operation.impl.generate.GenerateObjects;
+import uk.gov.gchq.gaffer.operation.impl.get.GetAdjacentIds;
+import uk.gov.gchq.gaffer.operation.impl.get.GetAllElements;
+import uk.gov.gchq.gaffer.operation.impl.get.GetElements;
+import uk.gov.gchq.gaffer.operation.impl.job.CancelScheduledJob;
+import uk.gov.gchq.gaffer.operation.impl.job.GetAllJobDetails;
+import uk.gov.gchq.gaffer.operation.impl.job.GetJobDetails;
+import uk.gov.gchq.gaffer.operation.impl.job.GetJobResults;
+import uk.gov.gchq.gaffer.operation.impl.join.Join;
+import uk.gov.gchq.gaffer.operation.impl.output.ToArray;
+import uk.gov.gchq.gaffer.operation.impl.output.ToCsv;
+import uk.gov.gchq.gaffer.operation.impl.output.ToEntitySeeds;
+import uk.gov.gchq.gaffer.operation.impl.output.ToList;
+import uk.gov.gchq.gaffer.operation.impl.output.ToMap;
+import uk.gov.gchq.gaffer.operation.impl.output.ToSet;
+import uk.gov.gchq.gaffer.operation.impl.output.ToSingletonList;
+import uk.gov.gchq.gaffer.operation.impl.output.ToStream;
+import uk.gov.gchq.gaffer.operation.impl.output.ToVertices;
+import uk.gov.gchq.gaffer.store.StoreException;
+import uk.gov.gchq.gaffer.store.StoreProperties;
+import uk.gov.gchq.gaffer.store.operation.GetSchema;
+import uk.gov.gchq.gaffer.store.operation.GetTraits;
+import uk.gov.gchq.gaffer.store.operation.add.AddSchemaToLibrary;
+import uk.gov.gchq.gaffer.store.operation.add.AddStorePropertiesToLibrary;
+import uk.gov.gchq.gaffer.store.operation.handler.OperationHandler;
+import uk.gov.gchq.gaffer.store.schema.Schema;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.UUID;
+
+/**
+ * An {@link AccumuloStore} that uses an Accumulo {@link MiniAccumuloCluster} to
+ * provide a {@link Connector}.
+ */
+public class MiniAccumuloStore extends AccumuloStore {
+    private static final String BASE_DIRECTORY = "miniAccumuloStoreTest-";
+    private static final String ROOTPW = "rootPW";
+    private MiniAccumuloCluster miniAccumuloCluster = null;
+    private MiniAccumuloConfig miniAccumuloConfig = null;
+    private Connector miniConnector;
+
+    public StoreProperties setUpTestDB(final StoreProperties properties) throws StoreException {
+        setProperties(properties);
+
+        File targetDir = new File("target");
```

Comment:
I can put it into the store properties via AccumuloProperties if you like.
I would avoid putting it into a subclass (e.g. MiniAccumuloProperties) as that will cause issues where tests are just supplied with properties and set up the (SingleUse)MiniAccumuloStore when it is required. This is because such tests currently lend themselves to being used with alternative properties including a 'real' and persistent 'AccumuloStore' rather than the Mini one for local test purposes.
I am wondering what sort of problem specifying the sub-directory name prefix as a property resolves, as the directory gets differentiated where more than one is required through the use of a UUID.randomUUID() suffix for each instance.

Reply:
In combination with the other suggestions, basically the benefit would be that you don't need this method which currently pollutes the store class.